pkg/queue/sharedmain/handlers.go

```diff
@@ -139,3 +146,13 @@ func withFullDuplex(h http.Handler, enableFullDuplex bool, logger *zap.SugaredLo
 		h.ServeHTTP(w, r)
 	})
 }
+
+func withRequestCounter(h http.Handler, pendingRequests *atomic.Int32) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get(network.ProbeHeaderName) != network.ProbeHeaderValue && !strings.HasPrefix(r.Header.Get("User-Agent"), "kube-probe/") {
```

Comment:
You can drop the probe checks - they should be handled by the drainer

https://github.com/knative/pkg/blob/main/network/handlers/drain.go#L154

Reply:
It was more so I could skip the counter on the healthchecks without modifying functionality in the external knative library. But I will remove